inst/stan/functions/observation_model.stan

```diff
@@ -51,22 +51,46 @@ void truncation_lp(array[] real truncation_mean, array[] real truncation_sd,
   }
 }
 // update log density for reported cases
-void report_lp(array[] int cases, vector reports,
+void report_lp(array[] int cases, array[] int cases_time, vector reports,
                array[] real rep_phi, real phi_mean, real phi_sd,
-               int model_type, real weight) {
+               int model_type, real weight, int accumulate) {
+  int n = num_elements(cases_time) - accumulate; // number of observations
+  vector[n] obs_reports; // reports at observation time
+  array[n] int obs_cases; // observed cases at observation time
+  if (accumulate) {
+    int t = num_elements(reports);
+    int i = 0;
+    int current_obs = 0;
+    obs_reports = rep_vector(0, n);
+    while (i <= t && current_obs <= n) {
+      if (current_obs > 0) { // first observation gets ignored when accumulating
+        obs_reports[current_obs] += reports[i];
+      }
+      if (i == cases_time[current_obs + 1]) {
+        current_obs += 1;
+      }
+      i += 1;
+    }
+    obs_cases = cases[2:(n + 1)];
+  } else {
+    obs_reports = reports[cases_time];
```

Comment:
I guess I am saying that cases_time would become optionally an array with a sum. Another option might be to make this a matrix multiplication problem

Reply:
Something like a matrix $M_{ij}$ which is 0 except if modelled observation $i$ gets added to true observation $j$? It's going to be quite sparse so do you think actually more efficient/ Or perhaps I'm misunderstanding.

And the array version would be the same, i.e. `cases_time` would be $C_i = j$ following the notation above?

Reply:
> It's going to be quite sparse so do you think actually more efficient? Or perhaps I'm misunderstanding.

Yes exactly and then handle it explicitly as a sparse matrix. I don't really know if it would be more efficient but my mind naturally likes to jump to the most complex possible solution :)

Reply:
> And the array version would be the same, i.e. cases_time would be 
 following the notation above?

Yes